index.js

```diff
@@ -0,0 +1,63 @@
+const crypto = require('crypto');
+const authMiddleware = require('express-basic-auth-safe');
+const debug = require('diagnostics')('authboot');
+
+module.exports = function (opts) {
+  opts = opts || {}
+  return function (app, options, callback) {
+    const users = new Map(
+      Object.entries(opts.users
+        || options.users
+        || app.config.get('auth:users')
+        || {}
+      )
+    );
+
+    const realm = opts.realm
+      || options.realm
+      || app.config.get('auth:realm');
+
+    const challenge = opts.challenge
+      || options.challenge
+      || app.config.get('auth:challenge');
+
+    const unauthorizedResponse = opts.errorPayload
+      || options.errorPayload
+      || app.config.get('auth:erroPayload')
+      || { error: 'Not authorized' }
+
+    const lookupOpt = typeof opts.lookup === 'function' ? opts.lookup : null;
```

Comment:
should the actual auth mechanism be separated out into another component and this just does some of the orchestration?

Reply:
in the future we could. This is currently just meant to be a convenience wrapper around hooking into the basic auth middleware that reduces the config boilerplate for integration purposes while giving some external flexibility for doing auth that is no longer dependent on encrypted config. I could rename this to `basic-authboot` to be more clear.